test/e2e/networking.go

```diff
@@ -101,157 +100,225 @@ var _ = Describe("Networking", func() {
 		}
 	})
 
-	//Now we can proceed with the test.
-	It("should function for intra-pod communication", func() {
-		// TODO: support DNS on vagrant #3580
-		SkipIfProviderIs("vagrant")
+	//1 service and 120 seconds is the original networking.go test,
+	//Adding another two elements (i.e. 5, 300) to the arrays will
+	//create a second test, which creates 5 parallel services, and
+	//each on a different port, and waits for all 5 to complete the
+	//pinging.
+	serviceSoakTests := []struct {
+		service int
+		timeout float64
+	}{
+		//These are very liberal, once this test is running regularly,
+		//We will DECREASE the timeout value.
+		{1, 300},
+		{2, 400}, //Test that two parallel nettests on 8080,8081 complete in 400 sec
+	}
 
-		By(fmt.Sprintf("Creating a service named %q in namespace %q", svcname, f.Namespace.Name))
-		svc, err := f.Client.Services(f.Namespace.Name).Create(&api.Service{
-			ObjectMeta: api.ObjectMeta{
-				Name: svcname,
-				Labels: map[string]string{
-					"name": svcname,
-				},
-			},
-			Spec: api.ServiceSpec{
-				Ports: []api.ServicePort{{
-					Protocol:   "TCP",
-					Port:       8080,
-					TargetPort: util.NewIntOrStringFromInt(8080),
-				}},
-				Selector: map[string]string{
-					"name": svcname,
-				},
-			},
-		})
-		if err != nil {
-			Failf("unable to create test service named [%s] %v", svc.Name, err)
-		}
+	// TODO: support DNS on vagrant #3580
+	for _, svcSoak := range serviceSoakTests {
+		timeout := svcSoak.timeout
+		serviceNum := svcSoak.service
```

Comment:
copy these variables into local ones to avoid   range iterator overwritingthe value in svcSoak which leads to a very confusing scenario.  Also its slightly more readable.